XenAdmin/SettingsPanels/CPUMemoryEditPage.cs

```diff
@@ -297,18 +384,16 @@ public AsyncAction SaveSettings()
 
             if (HasMemoryChanged)
             {
-                if (MROrGreater)
-                    actions.Add(memoryAction);  // Calculated in ValidToSave
-                else
-                    vm.Memory = Convert.ToInt64(this.nudMemory.Value * Util.BINARY_MEGA);
+                actions.Add(memoryAction);  // Calculated in ValidToSave
             }
 
             if (!Program.RunInAutomatedTestMode && vm.power_state != vm_power_state.Halted)
             {
-                if (!HasMemoryChanged)
-                    new ThreeButtonDialog(new ThreeButtonDialog.Details(SystemIcons.Information, Messages.VM_VCPU_CHANGES_NOT_SUPPORTED_MESSAGE, Messages.VM_LIVE_CHANGES_NOT_SUPPORTED_TITLE)).ShowDialog();
-                else if (!MROrGreater)
-                    new ThreeButtonDialog(new ThreeButtonDialog.Details(SystemIcons.Information, Messages.VM_VCPU_CHANGES_NOT_SUPPORTED_MESSAGE, Messages.VM_LIVE_CHANGES_NOT_SUPPORTED_TITLE)).ShowDialog();
+                if (HasVCPUChanged || HasTopologyChanged)

```

Comment:
We decided to disable the weight for running VMs, like we do with the other fields.